src/storage.rs

```diff
@@ -128,21 +232,20 @@ impl Storage {
     /// Reads data with given key to `Vec<u8>`, if error ocured or there are no
     /// records with matching key, returns `Err(_)`
     // @TODO specify more useful error type
-    pub fn read<K>(&mut self, key: &K) -> Result<ReadFuture>
-    where
-        K: AsRef<[u8]> + Ord,
-    {
-        if let Some(fut) = self
-            .active_blob
-            .as_ref()
-            .and_then(|active_blob| active_blob.read(&key).ok())
-        {
-            Ok(fut)
+    pub async fn read(&self, key: Vec<u8>) -> Result<Record> {
+        let inner = await!(self.inner.lock());
```

Comment:
no concurrent reads (

Reply:
Agree, RwLock from std needs some adaptation for async/await